From 6dfb27cc41932227bde0d6622cf6f33a1606bdbf Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Tue, 14 Mar 2023 05:52:14 +0200
Subject: [PATCH] [SW-120179] ppboot: gaudi2: change elbi axprot level

In secure mode, ELBI initiator need not be secured entity, instead
it can be user entity

Configure the ELBI AXPROT to user level in secured mode.
Leave ELBI AXPROT to privileged level in unsecured mode
which is the default value

Change-Id: I148bae4bbf9635316dd16d3d2b4565ca4d91ee54
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index d7c13f2..9067eed 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -773,8 +773,12 @@
 
 static void ppboot_secure_elbi(void)
 {
-	/* make transactions via ELBI as secured */
-	regmap_writel(HL_SEC_AXPROT_OVRD_SEC, mmPCIE_ELBI_RR_MSTR_IF_CORE_LBW_AXPROT_OVRD);
+	/*
+	 * Keep ELBI initiator as privileged in unsecured mode
+	 * which is the default value
+	 * In case of secure mode, override it as user
+	 */
+	regmap_writel(HL_SEC_AXPROT_USER_VAL, mmPCIE_ELBI_RR_MSTR_IF_CORE_LBW_AXPROT_OVRD);
 }
 
 #define SPI_DMA_AWPROT_PRIV		(1)
@@ -1220,7 +1224,6 @@
 
 void ppboot_security_init(void)
 {
-	ppboot_secure_elbi();
 	ppboot_secure_hbm();
 	ppboot_secure_spi_dma();
 
@@ -1230,6 +1233,7 @@
 		return;
 	}
 
+	ppboot_secure_elbi();
 	ppboot_secure_lbw();
 	ppboot_secure_hbw();
 }
